.github/workflows/pr-check-build-che-code-image.yaml

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (c) 2023 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+#
+
+name: Build che-code images Pull Request check
+
+# Trigger the workflow on pull request
+on:
+  pull_request:
+    paths-ignore:
+      - '**/*.md'
+      - devfile.yaml
+      - '.che/*'
+      - '.rebase/*'
+
+jobs:
+  build-che-code:
+    runs-on: ubuntu-22.04
+    steps:
+
+      - name: Checkout che-code source code
+        uses: actions/checkout@v2
+
+      - name: Login to Quay.io
+        uses: docker/login-action@v2
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_PULL_REQUESTS_USERNAME }}
+          password: ${{ secrets.QUAY_PULL_REQUESTS_PASSWORD }}
+
+      - name: Compile linux-libc
+        run: |
+          docker buildx build \
+            --platform linux/amd64 \
+            --progress=plain \
+            -f build/dockerfiles/linux-libc.Dockerfile \
```

Comment:
With the help of [matrix build](https://github.com/che-incubator/che-code/blob/4c94034316f9ea161fff00912fabbdd9ea653716/.github/workflows/pr-check.yml#L22), you can combine `Compile linux-libc` and `Compile linux-musl` steps into just a single step.